From ae1c1f8a7a1b31e25f160b55d25169abcce08a7c Mon Sep 17 00:00:00 2001
From: Jason Wu <huanyu@xilinx.com>
Date: Mon, 21 Jan 2013 17:20:30 +1000
Subject: [PATCH] arm: Makefile update to work with gnu-make 3.82

Split rule with both implicit targets and explicit targets into two rules.
One for implicit targets and one for explicit targets.
This is required to work with gnu make version 3.82 and above.

Signed-off-by: Jason Wu <huanyu@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm/Makefile |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 7a7bd8c..b7c0859 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -278,10 +278,6 @@ else
 KBUILD_IMAGE := zImage
 endif
 
-# With make 3.82 we cannot mix normal and wildcard targets
-BOOT_TARGETS1 := zImage Image xipImage bootpImage uImage
-BOOT_TARGETS2 := zImage% uImage.% dtbImage% simpleImage.%
-
 all:	$(KBUILD_IMAGE)
 
 boot := arch/arm/boot
@@ -292,8 +288,10 @@ archprepare:
 # Convert bzImage to zImage
 bzImage: zImage
 
-#zImage Image xipImage bootpImage uImage: vmlinux
-$(BOOT_TARGETS1) $(BOOT_TARGETS2) : vmlinux
+zImage Image xipImage bootpImage uImage: vmlinux
+	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $(boot)/$@
+
+zImage% uImage.% dtbImage% simpleImage.%: vmlinux
 	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $(boot)/$@
 
 zinstall uinstall install: vmlinux
-- 
1.7.1

